programs/loft.sc

```diff
@@ -0,0 +1,356 @@
+			p1=get(points,i);
+			c_for(t=0.05,t<1,t=t+0.05,
+				p2=get(points,i)*(1-t)^2+get(a,i)*2*t*(1-t)+get(points,i+1)*t^2;
+                if(p1!=p2,
+                    draw_shape('line',number_of_tick,{'from'->p1+0.5,'to'->p2+0.5})
+                );
+				p1=p2;
+			);
+			draw_shape('line',number_of_tick,{'from'->p1+0.5,'to'->get(points,i+1)+0.5})
+		)
+    )
+);
+
+_draw_surf(number_of_tick)->( //function used to render the surface created
+    m=length(global_cset);
+    n=length(global_cset:0);
+    if((m==1)||(n==1),
+        _draw_spline(global_points,number_of_tick)
+    );
+    if((m>1)&&(n>1),
+        c_for(j=0,j<m,j=j+1,
+            points=global_cset:j;
+            _draw_spline(points,number_of_tick)
+        );
+        c_for(i=0,i<n,i=i+1,
+            points=[];
+            c_for(j=0,j<m,j=j+1,
+                points+=global_cset:j:i;
+            );
+            _draw_spline(points,number_of_tick);
+        )
+    )
+);
+
+_set_block(pos)->( //function for setting block and storing their preavious value in global_affected_block
+    if(block(pos)!=global_block,
+        if((block(pos)==global_replace)||(global_replace==null),
+            existing_block=block(pos);
+	        global_affected_block+=[pos,existing_block];
+            set(pos,global_block)
+        )
+    )
+);
+
+_leng_approx(a,b,c)->( //function that approximate the length of a piece of quadratic spline given its 3 control points
+    L1=0.5*(a-b);
+    L2=0.5*(b-c);
+    L3=0.5*(a+b)-0.5*(b+c);
+    sqrt((L1:0)^2+(L1:1)^2+(L1:2)^2)+
+    sqrt((L2:0)^2+(L2:1)^2+(L2:2)^2)+
+    sqrt((L3:0)^2+(L3:1)^2+(L3:2)^2)
+);
+
+_paste_patch(p00,p01,p10,p11,a0,a1,b0,b1,c)->( //function used to paste a patch of the surface
+    v=0;
+    lvmax=max(
+        _leng_approx(p00,a0,p01),
+        _leng_approx(b0,c,b1),
+        _leng_approx(p10,a1,p11)
+    );
+    lumax=max(
+        _leng_approx(p00,b0,p10),
+        _leng_approx(a0,c,a1),
+        _leng_approx(p01,b1,p11)
+    );
+    if(lvmax>0, dv=0.7/lvmax, dv=1);
+    if(lumax>0, du=0.7/lumax, du=1);
+    while(v<=1,1000,
+        u=0;
+        while(u<=1,1000,
+            point=(p00*(1-v)^2+2*a0*v*(1-v)+p01*v^2)*(1-u)^2+     //point of the surface s
+                (2*b0*(1-v)^2+4*c*v*(1-v)+2*b1*v^2)*u*(1-u)+
+                (p10*(1-v)^2+2*a1*v*(1-v)+p11*v^2)*u^2;
+            p=[round(get(point,0)),round(get(point,1)),round(get(point,2))];
+            schedule(0,'_set_block',p);
+            if((u<1)&&(u+du>1),
+                u=1,
+                u=u+du
+            )
+        );
+        if((v<1)&&(v+dv>1),
+            v=1,
+            v=v+dv
+        )
+    )
+);
+
+paste(Block,replace)->(
+    global_block=Block;
+    global_replace=replace;
+    m=length(global_cset);
+    n=length(global_cset:0);
+    if((n>1)&&(m>1),
+        c_for(i=0,i<n-1,i=i+1,
+            game_tick(10);
```

Comment:
what was your intention with this line?

Reply:
Oh, ops. I forgot to delete line 317. I added the delay when I was trying to debug the script